internal/server.go

```diff
@@ -3,13 +3,21 @@ package internal
 import (
 	"encoding/xml"
 	"fmt"
+	"io"
 	"mime"
 	"net/http"
 	"net/url"
 	"strings"
 )
 
 func ServeError(w http.ResponseWriter, err error) {
+	if davErr, ok := err.(*DAVError); ok {
+		w.Header().Set("Content-Type", "text/xml; charset=utf-8")
+		w.WriteHeader(davErr.Code)
+		io.WriteString(w, davErr.Error())
```

Comment:
I think this is missing the XML header (`<?xml version="1.0" encoding="UTF-8"?>`).

We can just use `ServeXML` instead, which takes care of that + setting the Content-Type.

Reply:
Ah, the header is added in `Error`. Still, I'd prefer to use `ServeXML` here.